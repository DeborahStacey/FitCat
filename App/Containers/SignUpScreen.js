import React from 'react'
import { Alert, AsyncStorage, ScrollView, Text, TextInput, View } from 'react-native'
import { Actions as NavigationActions } from 'react-native-router-flux'
import I18n from 'react-native-i18n'
import { Colors } from '../Themes'
import RoundedButton from '../Components/RoundedButton'
import Dropdown from '../Components/Dropdown'
import { default as WellCatManager } from '../Services/WellCatManager'
import { default as StorageKeys } from '../Config/StorageKeys'

// Styles
import styles from './Styles/SignUpScreenStyle'

export default class WelcomeScreen extends React.Component {
  constructor () {
    super()
    this.state = {
      email: '',
      password: '',
      fName: '',
      lName: '',
      street: '',
      unit: '',
      city: '',
      pCode: '',
      locationId: -1,
      countries: {}
    }
  }

  componentWillMount () {
    var that = this
    WellCatManager.getCountries()
      .then((result) => {
        if (result.code === 1) {
          that.setState({
            countries: {'items': result.content}
          })
        } else {
          Alert.alert(`${I18n.t('tryLater')}`)
        }
      }).catch((error) => {
        Alert.alert(error)
      })
  }

  signUp (fName, lName, email, password, street, unit, city, pCode, locationId) {
    if (!unit) {
      unit = '-'
    }

    WellCatManager.signUp(fName, lName, email, password, street, unit, city, pCode, locationId)
      .then((result) => {
        if (result.code === 1) {
          AsyncStorage.setItem(StorageKeys.WELLCAT_EMAIL, email)
          NavigationActions.dashboard()
        } else if (result.code === 0) {
          Alert.alert(`${I18n.t('unableWellCat')} ${result.content}.`)
        } else if (result.code === -1) {
          Alert.alert(`${I18n.t('tryLater')}`)
        } else {
          Alert.alert(`${I18n.t('horrible')}`)
        }
      })
  }

  render () {
    return (
      <View style={styles.mainContainer}>
        <ScrollView style={styles.container}>
          <Text style={styles.sectionText} >
            {I18n.t('welcomeSignUp')}
          </Text>

          <View style={styles.section} >
            <Text style={styles.sectionText} >
              {I18n.t('fName')}
            </Text>
            <TextInput
              style={{height: 40, borderColor: 'gray', borderWidth: 1, paddingLeft: 10}}
              placeholder={I18n.t('fName')}
              placeholderTextColor={Colors.placeholderText}
              onChangeText={(fName) => this.setState({fName})}
              autoCapitalize={'none'}
            />
            <Text style={styles.sectionText} >
              {I18n.t('lName')}
            </Text>
            <TextInput
              style={{height: 40, borderColor: 'gray', borderWidth: 1, paddingLeft: 10}}
              placeholder={I18n.t('lName')}
              placeholderTextColor={Colors.placeholderText}
              onChangeText={(lName) => this.setState({lName})}
              autoCapitalize={'none'}
            />
            <Text style={styles.sectionText} >
              {I18n.t('email')}
            </Text>
            <TextInput
              style={{height: 40, borderColor: 'gray', borderWidth: 1, paddingLeft: 10}}
              placeholder={I18n.t('email')}
              placeholderTextColor={Colors.placeholderText}
              keyboardType={'email-address'}
              onChangeText={(email) => this.setState({email})}
              autoCapitalize={'none'}
            />
            <Text style={styles.sectionText} >
              {I18n.t('password')}
            </Text>
            <TextInput
              style={{height: 40, borderColor: 'gray', borderWidth: 1, paddingLeft: 10}}
              placeholder={I18n.t('password')}
              placeholderTextColor={Colors.placeholderText}
              onChangeText={(password) => this.setState({password})}
              autoCapitalize={'none'}
              secureTextEntry
            />
            <Text style={styles.sectionText} >
              {I18n.t('street')}
            </Text>
            <TextInput
              style={{height: 40, borderColor: 'gray', borderWidth: 1, paddingLeft: 10}}
              placeholder={I18n.t('street')}
              placeholderTextColor={Colors.placeholderText}
              onChangeText={(street) => this.setState({street})}
              autoCapitalize={'none'}
            />
            <Text style={styles.sectionText} >
              {I18n.t('unit')}
            </Text>
            <TextInput
              style={{height: 40, borderColor: 'gray', borderWidth: 1, paddingLeft: 10}}
              placeholder={I18n.t('unit')}
              placeholderTextColor={Colors.placeholderText}
              onChangeText={(unit) => this.setState({unit})}
              autoCapitalize={'none'}
            />
            <Text style={styles.sectionText} >
              {I18n.t('city')}
            </Text>
            <TextInput
              style={{height: 40, borderColor: 'gray', borderWidth: 1, paddingLeft: 10}}
              placeholder={I18n.t('city')}
              placeholderTextColor={Colors.placeholderText}
              onChangeText={(city) => this.setState({city})}
              autoCapitalize={'none'}
            />
            <Text style={styles.sectionText} >
              {I18n.t('pCode')}
            </Text>
            <TextInput
              style={{height: 40, borderColor: 'gray', borderWidth: 1, paddingLeft: 10}}
              placeholder={I18n.t('pCode')}
              placeholderTextColor={Colors.placeholderText}
              onChangeText={(pCode) => this.setState({pCode})}
              autoCapitalize={'none'}
            />
            <Text style={styles.sectionText} >
              {I18n.t('locationId')}
            </Text>
            <Dropdown
              options={this.state.countries}
              selectedValue={this.state.locationId}
              onValueChange={(locationId) => this.setState({locationId})}
            />
          </View>

          <View style={styles.paddedContainer}>
            <RoundedButton onPress={() => this.signUp(this.state.fName, this.state.lName, this.state.email, this.state.password, this.state.street, this.state.unit, this.state.city, this.state.pCode, this.state.locationId)}>
              {I18n.t('signUp')}
            </RoundedButton>
          </View>
        </ScrollView>
      </View>
    )
  }
}
