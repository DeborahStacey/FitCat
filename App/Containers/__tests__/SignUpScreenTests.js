/* eslint-disable no-global-assign, no-native-reassign */
import React, { Component } from 'react'
import SignUpScreen from '../SignUpScreen'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

jest.mock('Picker', () => {
  const React = require('React')
  return class MockPicker extends Component {
    static Item = props => React.createElement('Item', props, props.children)
    static propTypes = { children: React.PropTypes.any }

    render () {
      return React.createElement('Picker', this.props, this.props.children)
    }
  }
})

it('renders the sign up screen correctly', () => {
  fetch = jest.fn(() => new Promise(resolve => resolve()))
  const tree = renderer.create(
    <SignUpScreen />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
