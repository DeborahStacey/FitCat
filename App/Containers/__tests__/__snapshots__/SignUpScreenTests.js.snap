exports[`test renders the sign up screen correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F1F5FF",
      "flex": 1,
      "marginTop": 64,
    }
  }>
  <ScrollView
    style={
      Object {
        "flex": 1,
        "paddingTop": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontWeight": "bold",
          "marginVertical": 5,
          "textAlign": "center",
        }
      }>
      [missing \"en.welcomeSignUp\" translation]
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#D8D8D8",
          "borderBottomWidth": 1,
          "borderTopColor": "#D8D8D8",
          "borderTopWidth": 0.5,
          "margin": 25,
          "padding": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2C457F",
            "fontWeight": "bold",
            "marginVertical": 5,
            "textAlign": "center",
          }
        }>
        [missing \"en.fName\" translation]
      </Text>
      <TextInput
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="[missing \"en.fName\" translation]"
        placeholderTextColor="#787A7F"
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "paddingLeft": 10,
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2C457F",
            "fontWeight": "bold",
            "marginVertical": 5,
            "textAlign": "center",
          }
        }>
        [missing \"en.lName\" translation]
      </Text>
      <TextInput
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="[missing \"en.lName\" translation]"
        placeholderTextColor="#787A7F"
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "paddingLeft": 10,
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2C457F",
            "fontWeight": "bold",
            "marginVertical": 5,
            "textAlign": "center",
          }
        }>
        [missing \"en.email\" translation]
      </Text>
      <TextInput
        autoCapitalize="none"
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="[missing \"en.email\" translation]"
        placeholderTextColor="#787A7F"
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "paddingLeft": 10,
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2C457F",
            "fontWeight": "bold",
            "marginVertical": 5,
            "textAlign": "center",
          }
        }>
        [missing \"en.password\" translation]
      </Text>
      <TextInput
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="[missing \"en.password\" translation]"
        placeholderTextColor="#787A7F"
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "paddingLeft": 10,
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2C457F",
            "fontWeight": "bold",
            "marginVertical": 5,
            "textAlign": "center",
          }
        }>
        [missing \"en.street\" translation]
      </Text>
      <TextInput
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="[missing \"en.street\" translation]"
        placeholderTextColor="#787A7F"
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "paddingLeft": 10,
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2C457F",
            "fontWeight": "bold",
            "marginVertical": 5,
            "textAlign": "center",
          }
        }>
        [missing \"en.unit\" translation]
      </Text>
      <TextInput
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="[missing \"en.unit\" translation]"
        placeholderTextColor="#787A7F"
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "paddingLeft": 10,
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2C457F",
            "fontWeight": "bold",
            "marginVertical": 5,
            "textAlign": "center",
          }
        }>
        [missing \"en.city\" translation]
      </Text>
      <TextInput
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="[missing \"en.city\" translation]"
        placeholderTextColor="#787A7F"
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "paddingLeft": 10,
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2C457F",
            "fontWeight": "bold",
            "marginVertical": 5,
            "textAlign": "center",
          }
        }>
        [missing \"en.pCode\" translation]
      </Text>
      <TextInput
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="[missing \"en.pCode\" translation]"
        placeholderTextColor="#787A7F"
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "paddingLeft": 10,
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2C457F",
            "fontWeight": "bold",
            "marginVertical": 5,
            "textAlign": "center",
          }
        }>
        [missing \"en.locationId\" translation]
      </Text>
      <View
        style={
          Object {
            "borderColor": "grey",
            "borderWidth": 1,
            "flexDirection": "row",
            "flexWrap": "wrap",
            "zIndex": 99,
          }
        }>
        <Picker
          itemStyle={
            Object {
              "color": "black",
            }
          }
          onValueChange={[Function]}
          selectedValue={-1}
          style={
            Object {
              "flex": 95,
            }
          } />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 5,
              "paddingTop": 13,
            }
          } />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingBottom": 50,
          "paddingTop": 10,
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#466ECC",
            "borderRadius": 5,
            "height": 45,
            "justifyContent": "center",
            "marginHorizontal": 25,
            "marginVertical": 10,
            "opacity": 1,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginHorizontal": 5,
              "marginVertical": 10,
              "textAlign": "center",
            }
          }>
          [MISSING \"EN.SIGNUP\" TRANSLATION]
        </Text>
      </View>
    </View>
  </ScrollView>
</View>
`;
